main.tf

```diff
@@ -0,0 +1,82 @@
+ *
+ * [![Terraform validate](https://github.com/lablabs/terraform-aws-eks-keda/actions/workflows/validate.yaml/badge.svg)](https://github.com/lablabs/terraform-aws-eks-keda/actions/workflows/validate.yaml)
+ * [![pre-commit](https://github.com/lablabs/terraform-aws-eks-keda/actions/workflows/pre-commit.yaml/badge.svg)](https://github.com/lablabs/terraform-aws-eks-keda/actions/workflows/pre-commit.yaml)
+ */
+locals {
+  addon = {
+    name = "keda-controller"
+
+    helm_chart_name    = "keda"
+    helm_chart_version = "2.14.2"
+    helm_repo_url      = "https://kedacore.github.io/charts"
+  }
+
+  addon_irsa = {
+    operator = {
+      service_account_create = var.keda_operator_service_account_create
+      service_account_name   = var.keda_operator_service_account_name
+
+      irsa_role_create         = var.keda_operator_irsa_role_create
+      irsa_policy_enabled      = var.keda_operator_irsa_policy_enabled
+      irsa_policy              = var.keda_operator_irsa_policy
+      irsa_assume_role_enabled = var.keda_operator_irsa_assume_role_enabled
+      irsa_assume_role_arns    = var.keda_operator_irsa_assume_role_arns
+      irsa_additional_policies = var.keda_operator_irsa_additional_policies
+    }
+    metricServer = {
+      service_account_create = var.keda_metric_server_service_account_create
+      service_account_name   = var.keda_metric_server_service_account_name
+
+      irsa_role_create         = var.keda_metric_server_irsa_role_create
+      irsa_policy_enabled      = var.keda_metric_server_irsa_policy_enabled
+      irsa_policy              = var.keda_metric_server_irsa_policy
+      irsa_assume_role_enabled = var.keda_metric_server_irsa_assume_role_enabled
+      irsa_assume_role_arns    = var.keda_metric_server_irsa_assume_role_arns
+      irsa_additional_policies = var.keda_metric_server_irsa_additional_policies
+    }
+    webhooks = {
+      service_account_create = var.keda_webhooks_service_account_create
+      service_account_name   = var.keda_webhooks_service_account_name
+
+      irsa_role_create         = var.keda_webhooks_irsa_role_create
+      irsa_policy_enabled      = var.keda_webhooks_irsa_policy_enabled
+      irsa_policy              = var.keda_webhooks_irsa_policy
+      irsa_assume_role_enabled = var.keda_webhooks_irsa_assume_role_enabled
+      irsa_assume_role_arns    = var.keda_webhooks_irsa_assume_role_arns
+      irsa_additional_policies = var.keda_webhooks_irsa_additional_policies
+    }
+  }
+
+  addon_values = yamlencode({
+    rbac = {
+      create = var.rbac_create != null ? var.rbac_create : true
+    }
+    serviceAccount = {
+      operator = {
+        create = var.keda_operator_service_account_create
+        name   = var.keda_operator_service_account_name
+        annotations = length(module.addon-irsa["operator"].iam_role_attributes) > 0 ? {
+          "eks.amazonaws.com/role-arn" = module.addon-irsa["operator"].iam_role_attributes.arn
+        } : tomap({})
+      }
```

Comment:
I'm not sure if this condition won't cause some problems in combination with `tomap` when the argocd installation method is in use. We dealt with similar problems, for example, with this fix: https://github.com/lablabs/terraform-aws-eks-addon-template/commit/9d21a6cd019b6fe8722463538a5bd44faafe03bd. In my opinion, we should test this new IRSA mechanism for all installation methods, both when it's newly installed and when something is updated there.

Reply:
Reworked this using a new output variable `irsa_role_enabled` from the addon-irsa module